subprojects/model-core/src/test/groovy/org/gradle/api/internal/provider/DefaultPropertyTest.groovy

```diff
@@ -356,4 +356,16 @@ class DefaultPropertyTest extends PropertySpec<String> {
         1 * transformer.transform("abc") >> "cba"
         0 * _
     }
+
+    def "provider from property with convention can be absent"() {
+        def property = propertyWithDefaultValue(String)
+        def conventionValue = "convention"
+        property.convention(conventionValue)
```

Comment:
Nit: why the need for the local variable here?

Reply:
Inlined.